doc: update OpenThread RPC doc (#22588)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/ot_rpc.rst

API documentation
*****************

> This library mostly reimplements the existing OpenThread C API.

Reviewer: ```suggestion
This library is an implementation of the existing OpenThread C API.
```
I think this reads a bit better. "Reimplements" sounds a bit like we are disabling and enabling the OpenThread C API instead of having our own version of it.

Please refer to `OpenThread Reference`_ for the OpenThread C API documentation.

> Besides that, this library also implements a few API additions.

Reviewer: ```suggestion
In addition, the library also implements some API additions.
```
